src/Microsoft.DotNet.ProjectModel/Compilation/AnalyzerAssembly.cs

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using NuGet.Frameworks;
+
+namespace Microsoft.DotNet.ProjectModel.Compilation
+{
+    public class AnalyzerReference
+    {
+        /// <summary>
+        /// The fully-qualified path to the analyzer assembly.
+        /// </summary>
+        public string AssemblyPath { get; }
+        
+        /// <summary>
+        /// The supported language of the analyzer assembly.
+        /// </summary>
+        public string AnalyzerLanguage { get; }
+        
+        /// <summary>
+        /// The required framework for hosting the analyzer assembly.
+        /// </summary>
+        public NuGetFramework RequiredFramework { get; }
```

Comment:
@davidfowl (This description)[https://docs.nuget.org/create/analyzers-conventions] says:

>the specifiers in the path describe development host dependencies instead of build-time.

but I probably just don't know enough about lock files to understand what you're saying. Why would the analyzer care what the target TFM is? Or RID? If I'm building for OS X but the compiler is running on Linux, the only thing I should care about is whether or not the analyzer supports the Linux RID, right?